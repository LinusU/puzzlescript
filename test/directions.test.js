const { parseEngine } = require('./engine.test')

describe('Directions', () => {
    it('Marks a sprite when it wants to move', () => {
        const {engine, data} = parseEngine(`
title foo

========
OBJECTS
========

Background
green

Player
blue

=======
LEGEND
=======

. = Background
P = Player

================
COLLISIONLAYERS
================

Background
Player

===
RULES
===

[ Player ] -> [ > Player ]

=======
LEVELS
=======

P.

`)
        engine.tick()
        expect(engine.toSnapshot()).toMatchSnapshot()
        // Once these sprites actually move, we neet to separate engine.tick() into multiple steps:
        // 1. Update all the cells with new sprites and the wantsToMove directions
        // 2. Move all the sprites that want to move
        // 3. Late: Update all the cells with new sprites ...
        // 4. Late: Move all the sprites that want to move
        // next tick for all the AGAIN rules
        expect([...engine.currentLevel[0][0]._spriteAndWantsToMoves][0].b).toBe('RIGHT')

    })
})