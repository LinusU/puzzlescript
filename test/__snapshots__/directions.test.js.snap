// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directions Marks a sprite when it wants to move 1`] = `
Array [
  Array [
    Array [
      "Player",
    ],
    Array [
      "Background",
    ],
  ],
]
`;
